apps/prairielearn/src/migrations/20231031163354_client_fingerprints__create.sql

```diff
@@ -0,0 +1,30 @@
+CREATE TABLE
+  client_fingerprints (
+    id BIGSERIAL PRIMARY KEY,
+    user_id BIGINT REFERENCES users (user_id),
+    session_id TEXT REFERENCES user_sessions (session_id),
+    ip_address INET NOT NULL,
+    user_agent TEXT NOT NULL,
+    accept TEXT NOT NULL,
+    accept_language TEXT NOT NULL
+  );
+
+-- add client_fingerprint_id to user_sessions
+ALTER TABLE user_sessions
+ADD COLUMN client_fingerprint_id BIGINT REFERENCES client_fingerprints (id);
```

Comment:
@mwest1066 we had talked previously about whether or not we should ever destroy rows in the `user_sessions` table. This foreign key is perhaps a good argument to never do that? Alternatively, we could add `ON DELETE SET NULL` here.

Reply:
I'm currently leaning towards never deleting rows from `user_sessions`.